frontend/src/component/admin/apiToken/ApiTokenForm/TokenTypeSelector/TokenTypeSelector.tsx

```diff
@@ -7,45 +7,24 @@ import {
     RadioGroup,
     Typography,
 } from '@mui/material';
-import React from 'react';
-import { TokenType } from '../../../../../interfaces/token';
-import useUiConfig from '../../../../../hooks/api/getters/useUiConfig/useUiConfig';
-import { useOptionalPathParam } from '../../../../../hooks/useOptionalPathParam';
+import { TokenType } from 'interfaces/token';
 
 interface ITokenTypeSelectorProps {
     type: string;
-    setType: (value: string) => void;
+    setType: (value: TokenType) => void;
+    apiTokenTypes: {
+        key: string;
+        label: string;
+        title: string;
+        permission: string | string[];
+        hasAccess: boolean;
+    }[];
```

Comment:
We could maybe extract this type.

Reply:
Done in [6d67940](https://github.com/Unleash/unleash/pull/4072/commits/6d67940d64d9c7d24d44faf360d5fc9aad061b24)